R/derive_vars_extreme_date.R

```diff
@@ -0,0 +1,154 @@
+#' Derive Datetime of Any Extreme Dates
+#'
+#' Derives datetime of any extreme dates e.g. treatment start or end dates
+#'
+#' @param dataset Input dataset
+#'
+#' @param subject_keys Variables to uniquely identify a subject
+#'
+#'   The variables specified by the `subject_keys` parameter are expected.
+#'
+#'   Default: vars(STUDYID,USUBJID)
+#'
+#' @param dataset_date dataset that contains the date values e.g. `ex`
+#'
+#'   The variables `--ENDTC`, `--SEQ`, and those specified by the `filter_dataset_date`,
+#'   `dtc` and `subject_keys` parameters are expected.
+#'
+#' @param filter_dataset_date Filter condition for the dataset with dates
+#'
+#'   Only observations of the dataset which fulfill the specified condition
+#'   are considered for the treatment start date.
+#'
+#'   Default: NULL
+#'
+#'   Permitted Values: logical expression
+#'
+#' A list of quosures where the expressions are symbols as returned by
+#' `vars()` is expected.
+#'
+#' @param dtc name of variable from which the extreme date is derived e.g. `EXENDTC`
+#'
+#' @param new_var name of the extreme date derived e.g.treatment end date `TRTEDTM`
+#'
+#' @inheritParams impute_dtc
+#'
+#' @details For each group (with respect to the variables specified for the
+#'   `by_vars` parameter) the first observation (with respect to the order
+#'   specified for the `order` parameter) is included in the output dataset.
+#'
+#' @author Stefan Bundfuss, Teckla Akinyi
+#'
+#' @return The input dataset with extreme date variable added e.g. `TRTEDTM`
+#'
+#' @export
+#'
+#' @keywords adsl timing derivation
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#' data("ex")
+#' data("dm")
+#'
+#'dm %>%
+#'  derive_vars_extreme_date(subject_keys=vars(STUDYID,USUBJID),
+#'                           dataset_date = ex,
+#'                           filter_dataset_date = EXDOSE>0,
+#'                           dtc = EXSTDTC,
+#'                           order = vars(EXSTDTC,EXSEQ),
+#'                           new_var=TRTEDTM,
+#'                           date_imputation = "first",
+#'                           time_imputation = "first",
+#'                           flag_imputation = "AUTO",
+#'                           min_dates = NULL,
+#'                           max_dates = NULL,
+#'                           mode="LAST") %>%
+#'   select(USUBJID, TRTEDTM)
+derive_vars_extreme_date <- function(dataset,
+                                     subject_keys =vars(STUDYID,USUBJID),
+                                     dataset_date,
+                                     filter_dataset_date = NULL,
+                                     dtc,
+                                     order,
+                                     new_var,
```

Comment:
Should we replace `new_var` with `new_var_prefix`?

Reply:
ok